prime-router/src/main/kotlin/azure/EmailSenderFunction.kt

```diff
@@ -0,0 +1,140 @@
+import com.sendgrid.helpers.mail.objects.Personalization
+import gov.cdc.prime.router.secrets.SecretHelper
+import java.io.IOException
+import java.util.logging.Logger
+
+const val NO_REPLY_EMAIL = "no-reply@cdc.gov"
+const val REGISTER_EMAIL = "reportstream@cdc.gov"
+const val REGISTER_SUBJECT_BASE = "TOS Agreement for "
+const val TEMPLATE_ID = "d-472779cf554f418a9209acb62d2a48da"
+
+class SenderTosRequest {
+    data class SenderTosFormJSON(
+        val title: String,
+        val firstName: String,
+        val lastName: String,
+        val email: String,
+        val territory: String,
+        val organizationName: String,
+        val operatesInMultipleStates: Boolean,
+        val agreedToTermsOfService: Boolean
+    )
+}
+
+class EmailSenderFunction {
+
+    @FunctionName("emailRegisteredOrganization")
+    @StorageAccount("AzureWebJobsStorage")
+    fun emailRegisteredOrganization(
+        @HttpTrigger(
+            name = "emailRegisteredOrganization",
+            methods = [HttpMethod.POST],
+            authLevel = AuthorizationLevel.ANONYMOUS,
+            route = "email-registered"
+        )
+        request: HttpRequestMessage<String?>,
+        context: ExecutionContext,
+    ): HttpResponseMessage {
+        val logger: Logger = context.logger
+        var ret = request.createResponseBuilder(HttpStatus.BAD_REQUEST)
+
+        if (request.body !== null) {
+            logger.info(request.body)
+            ret.status(sendRegistrationConfirmation(request.body!!, logger))
+        }
+
+        return ret.build()
+    }
+
+    private fun sendRegistrationConfirmation(requestBody: String, logger: Logger): HttpStatus {
+        val gson = Gson()
+        val body = gson.fromJson<SenderTosRequest.SenderTosFormJSON>(
+            requestBody,
+            SenderTosRequest.SenderTosFormJSON::class.java
+        )
+        val title = body.title
+        val firstName = body.firstName
+        val lastName = body.lastName
+        val email = body.email
+        val territory = body.territory
+        val organizationName = body.organizationName
+        val operatesInMultipleStates = body.operatesInMultipleStates
+        val agreedToTermsOfService = body.agreedToTermsOfService
+
+        var status: HttpStatus = HttpStatus.NOT_FOUND
+        val content: Content = Content()
+        content.type = "plain/text"
+
+        val p: Personalization = Personalization()
+        p.addTo(Email(REGISTER_EMAIL))
+        p.addCc(Email(email))
+        p.addDynamicTemplateData(
+            "formData",
+            object {
+                var title = title
+                var firstName = firstName
+                var lastName = lastName
+                var email = email
+                var territory = territory
+                var organizationName = organizationName
+                var operatesInMultipleStates = operatesInMultipleStates
+                var agreedToTermsOfService = agreedToTermsOfService
+            }
```

Comment:
Ah, I'd been considering it all from the context of a form sending the data. There should definitely be more done to prevent this stuff if someone were to programatically call the API.

I'll start working on the refactor of this file tomorrow and keep in mind the following concerns:

String validators (regex should do?)
Double checking to ensure our gson implementation prevents wrong types
Ensuring submission spam cannot occur via front-end form

Reply:
- [X] Gson will handle ensuring our data maintains its structural integrity according to our `data class` model (including type).
- [X] Non-null validator functions have been added to the `data class` to ensure no required values are null and the agreement boolean is true. No mail is built/sent otherwise.

I believe the last two concerns from this comment left to address are ensuring:
- [ ] `nonce` on front-end to ensure no form re-submission and spam
- [ ] body values exceeding a max-length (both front-end and back-end checks)